src/main/scala/io/iohk/ethereum/network/p2p/messages/BaseETH6XMessages.scala

```diff
@@ -134,26 +158,76 @@ object BaseETH6XMessages {
     override def code: Int = Codes.NewBlockCode
   }
 
-  object SignedTransactions {
+  object TypedTransaction {
+    implicit class TypedTransactionsRLPAggregator(val encodables: Seq[RLPEncodeable]) extends AnyVal {
 
-    lazy val chainId: Byte = Config.blockchains.blockchainConfig.chainId
+      /** Convert a Seq of RLPEncodable containing TypedTransaction informations into a Seq of
+        * Prefixed RLPEncodable.
+        *
+        * PrefixedRLPEncodable(prefix, prefixedRLPEncodable) generates binary data
+        * as prefix || RLPEncodable(prefixedRLPEncodable).
+        *
+        * As prefix is a byte value lower than 0x7f, it is read back as RLPValue(prefix),
+        * thus PrefixedRLPEncodable is binary equivalent to RLPValue(prefix), RLPEncodable
+        *
+        * The method aggregates back the typed transaction prefix with the following heuristic:
+        * - a RLPValue(byte) with byte < 07f + the following RLPEncodable are associated as a PrefixedRLPEncodable
+        * - all other RLPEncodable are kept unchanged
+        *
+        * This is the responsibility of the RLPDecoder to insert this meaning into its RLPList, when appropriate.
+        *
+        * @return a Seq of TypedTransaction enriched RLPEncodable
+        */
+      def toTypedRLPEncodables: Seq[RLPEncodeable] =
+        encodables match {
+          case Seq() => Seq()
+          case Seq(RLPValue(v), rlpList: RLPList, tail @ _*) if v.length == 1 && v.head < 0x7f =>
```

Comment:
Do you think it would make sense not to repeat the magic `0x7f` in multiple places but putting it for example in a wrapper value class for `Byte`?

Reply:
good idea, I've extracted the magic number and added the proposed wrapper